flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/functions/CollectionFunctionsITCase.java

```diff
@@ -0,0 +1,261 @@
+import java.util.Map;
+
+import static org.apache.flink.table.api.Expressions.$;
+import static org.apache.flink.table.api.Expressions.array;
+import static org.apache.flink.table.api.Expressions.call;
+import static org.apache.flink.table.api.Expressions.map;
+import static org.apache.flink.table.api.Expressions.mapFromArrays;
+import static org.apache.flink.table.api.Expressions.row;
+
+/** Tests for collections {@link BuiltInFunctionDefinitions}. */
+public class CollectionFunctionsITCase extends BuiltInFunctionTestBase {
+
+    @Parameterized.Parameters(name = "{index}: {0}")
+    public static List<TestSpec> testData() {
+        return Arrays.asList(
+                TestSpec.forFunction(BuiltInFunctionDefinitions.MAP_KEYS, "Null inputs")
+                        .onFieldsWithData(null, null, "item")
+                        .andDataTypes(
+                                DataTypes.BOOLEAN().nullable(),
+                                DataTypes.INT().nullable(),
+                                DataTypes.STRING())
+                        .testTableApiError(
+                                call("MAP_KEYS", $("f0"), $("f1")),
+                                "Invalid function call:\nMAP_KEYS(BOOLEAN, INT)")
+                        .testResult(
+                                map(
+                                                $("f0").cast(DataTypes.BOOLEAN()),
+                                                $("f1").cast(DataTypes.INT()))
+                                        .mapKeys(),
+                                "MAP_KEYS(map[cast(f0 as boolean), cast(f1 as int)])",
+                                new Boolean[] {null},
+                                DataTypes.ARRAY(DataTypes.BOOLEAN()))
+                        .testTableApiResult(
+                                map($("f2"), $("f1").cast(DataTypes.INT())).mapKeys(),
+                                new String[] {"item"},
+                                DataTypes.ARRAY(DataTypes.STRING()))
+                        .testSqlResult(
+                                "MAP_KEYS(map[cast(NULL as int), 'value'])",
+                                new Integer[] {null},
+                                DataTypes.ARRAY(DataTypes.INT())),
+                TestSpec.forFunction(BuiltInFunctionDefinitions.MAP_KEYS)
+                        .onFieldsWithData(1, "one", 2, "two")
+                        .andDataTypes(
+                                DataTypes.INT(),
+                                DataTypes.STRING(),
+                                DataTypes.INT(),
+                                DataTypes.STRING())
+                        .testResult(
+                                map($("f0"), $("f1"), $("f2"), $("f3")).mapKeys(),
+                                "MAP_KEYS(map[f0, f1, f2, f3])",
+                                new Integer[] {1, 2},
+                                DataTypes.ARRAY(DataTypes.INT()))
+                        .testTableApiResult(
```

Comment:
we are executing a lot of Flink jobs under the hood, of course testing is important but they should be well chosen, e.g. we could remove this one because it is very similar to the one above

Reply:
I think three to four tests for `mapKeys` should be enough no?

Reply:
yes, seems it is possible to decrease number of tests.
I made it in the next commit